### PR TITLE
Sharing federated tokens in oidc authentication

### PR DESCRIPTION
**Pull request by indeewari**
### Proposed changes in this pull request

Passing the federated token data from oidc authenticator to the oauth side

-

### When should this PR be merged

Once reviewed.

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package org.wso2.carbon.identity.application.authentication.framework.model;
```
Comment (pamodaaw) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:1]: Missing license header

Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:12]: Fixed 3d3d35eee8e1fc89be30973a127eb9f800c9edc3

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * This class is model class of a federated token.
+ * A federated token is an external token obtained via an OIDC federated authenticator
+ * after a successful authentication.
+ */
+public class FederatedToken implements Serializable {
+
+    private static final long serialVersionUID = 6618332057931299623L;
+    private String idp;
+    @JsonProperty("expires_in")
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:28]: Lets not use this class to directly build the response in the oauth level

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return refreshToken;
+    }
+
+    public void setRefreshToken(String refreshToken) {
+
+        this.refreshToken = refreshToken;
+    }
+
+    @Override
+    public String toString() {
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:95]: Lets remove this method as someone could unknowingly use this in a log and tokens could be logged.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 4]
```diff
@@ -685,6 +686,11 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
             publishAuthenticationSuccess(request, context, sequenceConfig.getAuthenticatedUser());
         }
 
+        // Passing the federated tokens to the authentication result.
+        if (context.getProperty(FEDERATED_TOKENS) != null) {
+            authenticationResult.addProperty(FEDERATED_TOKENS, context.getProperty(FEDERATED_TOKENS));
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:691]: Lets add a debug log here mentioning federated tokens are available and being added to the auth result. Would be better if we can mention from which IDPs as well.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 5]
```diff
@@ -685,6 +686,11 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
             publishAuthenticationSuccess(request, context, sequenceConfig.getAuthenticatedUser());
         }
 
+        // Passing the federated tokens to the authentication result.
+        if (context.getProperty(FEDERATED_TOKENS) != null) {
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:690]: Lets add unit test to make sure that available federated tokens are getting added to the authresult

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 4 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:691]: Fixed cfb085b712df445a0a612df800c012b16fa75faf

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java (hunk 2 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java]: Fixed cfb085b712df445a0a612df800c012b16fa75faf

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java (hunk 3 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java]: Fixed cfb085b712df445a0a612df800c012b16fa75faf

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 LLC. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content.
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:8]: Add the license

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+public class FederatedToken implements Serializable {
+
+    private static final long serialVersionUID = 6618332057931299623L;
+    private String idp;
+    private String tokenValidityPeriod;
+    private String scope;
+    private String accessToken;
+    private String refreshToken;
+
+    // Constructor
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:30]: Add the javadoc comment

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 8]
```diff
@@ -685,6 +686,17 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
...
+            authenticationResult.addProperty(FEDERATED_TOKENS, context.getProperty(FEDERATED_TOKENS));
+            if (log.isDebugEnabled()) {
+                log.debug("Federated tokens are available in the authentication context for the IDP:" +
+                        FrameworkUtils.getFederatedAuthenticatorName(context));
+                log.debug("Federated tokens are added to the authentication result. IDP:" +
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:695]: Add to a single log

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 9]
```diff
@@ -4154,4 +4154,22 @@ public static String getProcessedClaimValue(String claimURI, String claimValue,
...
     }
+
+    /**
+     * This method returns the current federated authenticator name. If there is no external IdP, then the current
+     * authenticator name is returned.
+     *
+     * @param context Authentication context.
+     * @return Federated authenticator name.
+     */
+    public static String getFederatedAuthenticatorName(AuthenticationContext context) {
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java:4165]: Add this as a private method

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 10]
```diff
@@ -4154,4 +4154,22 @@ public static String getProcessedClaimValue(String claimURI, String claimValue,
...
+
+        if (context == null) {
+            return StringUtils.EMPTY;
+        }
+        if (context.getExternalIdP() == null) {
+            return context.getCurrentAuthenticator();
+        }
+        return context.getExternalIdP().getIdPName();
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java:4173]: There can be multiple IDPs, log them all

Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java:4173]: use federated token property

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 11]
```diff
@@ -103,6 +103,7 @@
 public class DefaultAuthenticationRequestHandler implements AuthenticationRequestHandler {
 
     public static final String AUTHZ_FAIL_REASON = "AUTHZ_FAIL_REASON";
+    public static final String FEDERATED_TOKENS = "federated_tokens";
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:106]: Use the framework constant and use it from there.

Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:106]: Use the same for all modules

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+ * This class is model class of a federated token.
+ * A federated token is an external token obtained via an OIDC federated authenticator
+ * after a successful authentication.
+ */
+public class FederatedToken implements Serializable {
+
+    private static final long serialVersionUID = 6618332057931299623L;
+    private String idp;
+    private String tokenValidityPeriod;
+    private String scope;
```
Comment (indeewari) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:26]: add a comment to the variable space delimtted multiple scopes

Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java:34]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java (hunk 6 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java (hunk 7 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/FederatedToken.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 8 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java (hunk 9 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java (hunk 10 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 11 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: Fixed ed7acde27e7e2af5be270a50e68fbf3c9d77141e

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 13]
```diff
@@ -120,6 +122,27 @@ public static DefaultAuthenticationRequestHandler getInstance() {
...
     }
 
+    /**
+     * This method returns the comma separated list of federated authenticator names bounded to the
+     * federated_tokens property in the authentication context.
+     *
+     * @param context Authentication context containing the federated_tokens property.
+     * @return Comma separated list of the federated authenticator names.
+     */
+    public static String getFederatedAuthenticatorName(AuthenticationContext context) {
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:132]: make this private

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 14]
```diff
@@ -105,6 +106,7 @@ public class DefaultAuthenticationRequestHandler implements AuthenticationReques
     public static final String AUTHZ_FAIL_REASON = "AUTHZ_FAIL_REASON";
     private static final Log log = LogFactory.getLog(DefaultAuthenticationRequestHandler.class);
     private static final Log AUDIT_LOG = CarbonConstants.AUDIT_LOG;
+    public static final String COMMA = ",";
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:109]:
```suggestion
    private static final String COMMA = ",";
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 13 above)
Comment (janakamarasena) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:132]: since this util is specific to the federated tokens, lets pass in the federated tokens list here

Comment (janakamarasena) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:132]: also lets move this util method to the end of the class

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 15]
```diff
@@ -685,6 +708,16 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
             publishAuthenticationSuccess(request, context, sequenceConfig.getAuthenticatedUser());
         }
 
+        // Passing the federated tokens to the authentication result.
+        if (context.getProperty(FrameworkConstants.FEDERATED_TOKENS) != null) {
+            authenticationResult.addProperty(
+                    FrameworkConstants.FEDERATED_TOKENS, context.getProperty(FrameworkConstants.FEDERATED_TOKENS));
+            if (log.isDebugEnabled()) {
+                log.debug("Federated tokens are available in the authentication context for the IDP:" +
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java:716]:
```suggestion
                log.debug("Federated tokens are available in the authentication context for the IDP: " +
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 16]
```diff
@@ -255,6 +255,8 @@ public abstract class FrameworkConstants {
     public static final String IDP_MAPPED_USER_ROLES = "identityProviderMappedUserRoles";
     public static final String ALLOW_ASSOCIATING_TO_EXISTING_USER = "JITProvisioning.AllowAssociatingToExistingUser";
 
+    // The constant to used as the attribute key or the property key of the federated tokens.
+    public static final String FEDERATED_TOKENS = "federated_tokens";
     private FrameworkConstants() {
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:260]: lets add a new line here

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 17]
```diff
@@ -4154,4 +4154,5 @@ public static String getProcessedClaimValue(String claimURI, String claimValue,
             throw new FrameworkException("Error occurred while encrypting claim value of: " + claimURI, e);
         }
     }
+
```
Comment (janakamarasena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java:4157]: shall we revert this change as its irrelevant?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 13 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: Fixed a0f882011c467a8f0c781821bfcd6ad8f03da0c5

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 14 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: Fixed a0f882011c467a8f0c781821bfcd6ad8f03da0c5

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java (hunk 15 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: Fixed a0f882011c467a8f0c781821bfcd6ad8f03da0c5

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java (hunk 16 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:261]: Fixed a0f882011c467a8f0c781821bfcd6ad8f03da0c5

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java (hunk 17 above)
Comment (indeewari) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Fixed a0f882011c467a8f0c781821bfcd6ad8f03da0c5

**Review by janakamarasena — APPROVED** [review]

**Review by janakamarasena — APPROVED** [review]
